[pytorch] [caffe2] Add support for converting quantized AvgPool2d and Reshape operations

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #30498 [pytorch][caffe2] Add support for quantized slice conversion
* **#30490 [pytorch] [caffe2] Add support for converting quantized AvgPool2d and Reshape operations**

Summary:
Add symbolic mapping to Int8AvgPool2d and Int8Reshape op in C2

Test Plan:

python test/onnx/test_pytorch_onnx_caffe2_quantized.py TestQuantizedOps
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D18740520](https://our.internmc.facebook.com/intern/diff/D18740520)

supriyar committed:
[pytorch] [caffe2] Add support for converting quantized AvgPool2d and Reshape operations

Summary:
Add symbolic mapping to Int8AvgPool2d and Int8Reshape op in C2

Test Plan:

python test/onnx/test_pytorch_onnx_caffe2_quantized.py TestQuantizedOps
Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

houseroad commented on caffe2/onnx/backend.cc:
```diff
@@ -1361,6 +1362,17 @@ Caffe2Ops Caffe2Backend::CreateLRN(
   return c2_op;
 }
 
+Caffe2Ops Caffe2Backend::CreateInt8Reshape(
```
Can we share the code between CreateReshape and CreateInt8Reshape instead of copying it?

houseroad commented on test/onnx/test_pytorch_onnx_caffe2_quantized.py:
```diff
@@ -7,14 +7,14 @@
 
 import onnx
 import caffe2.python.onnx.backend as c2
+torch.backends.quantized.engine = "qnnpack"
```
Nit: move setting into the test setup stage?

houseroad commented on torch/onnx/symbolic_caffe2.py:
```diff
@@ -13,11 +13,35 @@ def register_quantized_ops(domain, version):
                 sym_registry.register_op(op[0], op[1], '', version)
             sym_registry.register_op(op[0], op[1], domain, version)
 
+def nchw2nhwc(g, input):
```
most of the logic of nchw2nhwc is the same, let's lift the common part out.

houseroad commented on torch/onnx/symbolic_caffe2.py:
```diff
@@ -130,7 +154,41 @@ def upsample_nearest2d(g, input, output_size, align_corners=None):
+        "Y_scale_f": input.node()["Y_scale"],
+        "Y_zero_point_i": input.node()["Y_zero_point"],
+    }
+    print("shape is ", shape)
```
move debug info?

houseroad reviewed (commented):
The overall logic looks good. Just need to polish the PR a bit more.

supriyar committed:
Update on "[pytorch] [caffe2] Add support for converting quantized AvgPool2d and Reshape operations"


Summary:
Add symbolic mapping to Int8AvgPool2d and Int8Reshape op in C2

Test Plan:

python test/onnx/test_pytorch_onnx_caffe2_quantized.py TestQuantizedOps
Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

houseroad reviewed (approved):
Looks good.

supriyar committed:
Update on "[pytorch] [caffe2] Add support for converting quantized AvgPool2d and Reshape operations"


Summary:
Add symbolic mapping to Int8AvgPool2d and Int8Reshape op in C2

Test Plan:

python test/onnx/test_pytorch_onnx_caffe2_quantized.py TestQuantizedOps
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D18740520](https://our.internmc.facebook.com/intern/diff/D18740520)

[ghstack-poisoned]
